matchmaking/lobby/lobby.gd

```diff
@@ -57,12 +57,21 @@ func launch_game():
 	], false, HTTPClient.METHOD_POST)
 
 func check_ready_state():
+	get_node("GUI/Body/Footer/LaunchButton").disabled = ! is_ready_state()
+
+func is_ready_state():
 	if Store._state.lobby.creator.id != Store._state.player.id || Store._state.lobby.players.size() < 2:
-		return
+		return false
+	var faction_in_lobby = []
 	for player in Store._state.lobby.players:
 		if player.ready == false:
-			return
-	get_node("GUI/Body/Footer/LaunchButton").disabled = false
+			return false
+		if player.faction != null:
+			if not faction_in_lobby.has(int(player.faction)): # it has to be cast as an int to work
```

Comment:
fusionner avec le if du dessus pour éviter les if imbriqués. à l'aide d'un &&.

transformer not en !